panel/lxqtpanel.cpp

```diff
@@ -374,7 +372,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate)
+void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
```

Comment:
OK, I couldn't test your last suggestion because of the wrong diff but I think the top and left panel animation shouldn't work with it.

You're stepping into the same pitfalls I did when I started this ;) That's why I wrote, in the PR comment, "That was the best solution I found to several problems..."

I think your previous suggestion is the best compromise.